contracts/Tokens/Prime/Prime.sol

```diff
@@ -0,0 +1,568 @@
+     * @param market the market vToken address
+     * @return capital the capital to use in calculation of score
+     */
+    function _capitalForScore(
+        uint256 xvs,
+        uint256 borrow,
+        uint256 supply,
+        address market
+    ) internal view returns (uint256) {
+        uint256 borrowCap = (xvs * markets[market].borrowMultiplier) / EXP_SCALE;
+        uint256 supplyCap = (xvs * markets[market].supplyMultiplier) / EXP_SCALE;
+
+        if (supply > supplyCap) {
+            supply = supplyCap;
+        }
+
+        if (borrow > borrowCap) {
+            borrow = borrowCap;
+        }
+
+        return (supply + borrow);
+    }
+
+    /**
+     * @notice Used to mint a new prime token
+     * @param isIrrevocable is the tokens being issued is irrevocable
+     * @param owner token owner
+     */
+    function _mint(bool isIrrevocable, address owner) internal {
+        require(tokens[owner].exists == false, "user already owns a prime token");
+
+        tokens[owner].exists = true;
+        tokens[owner].isIrrevocable = isIrrevocable;
+
+        if (isIrrevocable == true) {
+            _totalIrrevocable++;
+        } else {
+            _totalRevocable++;
+        }
+
+        require(
+            _totalIrrevocable <= _irrevocableLimit && _totalRevocable <= _revocableLimit,
+            "exceeds token mint limit"
+        );
+
+        _updateRoundAfterTokenMinted();
+
+        emit Mint(owner, isIrrevocable);
+    }
+
+    /**
+     * @notice Used to burn a new prime token
+     * @param owner owner whose prime token to burn
+     */
+    function _burn(address owner) internal {
+        require(tokens[owner].exists == true, "user doesn't own an prime token");
+
+        tokens[owner].exists = false;
+        tokens[owner].isIrrevocable = false;
+
+        if (tokens[owner].isIrrevocable == true) {
+            _totalIrrevocable--;
+        } else {
+            _totalRevocable--;
+        }
+
+        _updateRoundAfterTokenBurned();
+
+        emit Burn(owner);
+    }
+
+    /**
+     * @notice Used to get if the XVS balance is eligible for prime token
+     * @param amount amount of XVS
+     */
+    function isEligible(uint256 amount) internal view returns (bool) {
+        if (amount >= MINIMUM_STAKED_XVS) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @notice Accrue rewards for the user. Must be called by Comptroller before changing account's borrow or supply balance.
+     * @param account account for which we need to accrue rewards
+     * @param vToken the market for which we need to accrue rewards
+     */
+    function executeBoost(address account, address vToken) public {
+        if (markets[vToken].exists == false) {
+            return;
+        }
+
+        if (tokens[account].exists == false) {
+            return;
+        }
+
+        accrueInterest(vToken);
+        interests[vToken][account].accrued = _interestAccrued(vToken, account);
```

Comment:
Do we need to store this value? We are not using it at any place (only to set it to zero after claiming). Maybe we can transform the `_interestAccrued` internal view into a public view